src/experimental/tools/download2go/apis/dash/dashConnectivity.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  IRepresentation,
+  ISegmentsBuiltType,
+  ISegmentBuilder,
+  SegmentBuilt,
+} from "./types";
+import { ISegment } from "../../../../../manifest";
+import { ISidxSegment } from "../../../../../parsers/containers/isobmff";
+import parseManifest from "../../../../../parsers/manifest/dash/index";
+import { IParserResponse } from "../../../../../parsers/manifest/dash/parse_mpd";
+import { IParsedManifest } from "../../../../../parsers/manifest/types";
+import { IUtils } from "./../../types";
+import { TypedArray } from "../drm/keySystems";
+
+/**
+ * Get a manifest from a server and parse it.
+ *
+ * @param url - The url manifest mpd file
+ * @returns The manifest parsed IParserResponse<IParsedManifest>
+ *
+ */
+export const getOnlineMPDParsed = async (
+  url: string
+): Promise<IParserResponse<IParsedManifest>> => {
+  const data = await makeHTTPRequest<string>(url, {
+    method: "GET",
+    responseType: "text",
+  });
+  return parseManifest(new DOMParser().parseFromString(data, "text/xml"), {
+    loadExternalClock: false,
+    url,
+  });
+};
+
+/**
+ * Build the init segments to be able to retieve SIDX (index/init)
+ *
+ * @param IRepresentation - The segmentBase and initialization segment
+ * @returns The initSegment and indexSegment
+ *
+ */
+export const buildInitIndexSegment = async ({
+  segmentBase,
+  initialization,
+}: IRepresentation): Promise<{
+initSegment: TypedArray | ArrayBuffer;
+indexSegment: TypedArray | ArrayBuffer;
+}> => {
+  try {
+    if (!initialization.mediaURL) {
+      throw new Error("MediaURL from the initialization segment is broken");
+    }
+    const [initSegment, indexSegment] = await Promise.all([
+      makeHTTPRequest<TypedArray | ArrayBuffer>(initialization.mediaURL, {
+        headers: { Range: `bytes=${initialization.range.join("-")}` },
+        method: "GET",
+        responseType: "arraybuffer",
+      }),
+      makeHTTPRequest<TypedArray | ArrayBuffer>(initialization.mediaURL, {
+        headers: { Range: `bytes=${segmentBase.indexRange.join("-")}` },
+        method: "GET",
+        responseType: "arraybuffer",
+      }),
+    ]);
+    return { initSegment, indexSegment };
+  } catch (e) {
+    throw new SegmentConstuctionError(e.message);
+  }
+};
+
+/**
+ * Get a segment from a server.
+ *
+ * @param ISidxSegment - SIDX or ISegment Segment information
+ * @param url - base url of the segment
+ * @returns The data,duration,time,timescale of the current segment retrieved
+ *
+ */
+export const getSegmentBuffer = async ({
+  segment,
+  url,
+  type,
+}: {
+segment: ISidxSegment | ISegment;
+url: string | null;
+type: "TemplateRepresentationIndex" | "BaseRepresentationIndex";
+}): Promise<ISegmentsBuiltType> => {
+  try {
+    if (type === "BaseRepresentationIndex") {
+      if (!url) {
+        throw new SegmentConstuctionError(
+          "The mediaURL is not defined for the given representation segment"
+        );
```

Comment:
So you throw a SegmentConstructionError in a try so you can re-wrap it in a SegmentConstructionError?
I think I understand your problem here but it is still odd to read